### ISSUE TITLE
proxy demo failed when sending file.

### ISSUE THREAD

**Issue by xudesheng** [bk-rs/ssh-rs#13]
it works perfectly when executing the remote commands.

But once I add some code to upload files, it works for the small files but failed when uploading a large file:
```
        //new code
        let local_file_path = Path::new("/home/dxu/async-ssh2-lite/target/debug/proxy.d");
        let remote_file_path = Path::new("/home/dxu/proxy.d");
        let metadata = std::fs::metadata(local_file_path)?;
        let size = metadata.len();
        println!("file size:{}", size);
        let mut buf:Vec<u8> = vec![0; size as usize];
        let mut f = std::fs::File::open(local_file_path)?;
        f.read_to_end(&mut buf)?;

        // std::fs::read(local_file_path, &mut buf)?;
        let mut remote_file = session.scp_send(remote_file_path,0o644, size, None).await?;
        remote_file.write_all(&buf).await?;
        remote_file.flush().await?;
        remote_file.send_eof().await?;
        remote_file.wait_eof().await?;
        remote_file.close().await?;
        remote_file.wait_close().await?;

        //new code
        let mut channel = session.channel_session().await?;
```
for a small file, it uploaded successfully. (size:334)
for a large file, it failed (local size: 27704336, remote size:2572288)
```
bastion_channel read 0
task_with_main run failed, err Custom { kind: Other, error: "Failure while draining incoming flow" }
```

**Comment by xudesheng** [comment]
I also tried with:
```
let mut remote_file = session.scp_send(remote_file_path,0o644, size, None).await?;
        let mut total_write = 0;
        while total_write < size {
            let n = remote_file.write(&buf[total_write as usize ..]).await?;
            total_write += n as u64;
            if n == 0 {
                break;
            }
        }
        // remote_file.write_all(&buf).await?;
        remote_file.flush().await?;
```
or
```
let sftp = session.sftp().await?;
        let mut remote_file = sftp.create(remote_file_path).await?;
        remote_file.write_all(&buf).await?;
        remote_file.flush().await?;
        // remote_file.send_eof().await?;
        // remote_file.wait_eof().await?;
        remote_file.close().await?;
        // remote_file.wait_close().await?;
```
the error is the same.

the solution mentioned in: https://github.com/bk-rs/async-ssh2-lite/issues/10 seems like have the same issue.
@vkill appreciate you can take a look. it is a block issue to me now.

**Comment by xudesheng** [comment]
compare the log on the server side:
if I use Python `paramiko`, it succeeded, the log:
```
Nov 21 02:57:49 twx-0 sshd[3912]: Starting session: command for dxu from 10.133.1.12 port 55398 id 0
Nov 21 02:57:52 twx-0 sshd[3912]: debug1: Received SIGCHLD.
Nov 21 02:57:52 twx-0 sshd[3912]: debug1: session_by_pid: pid 3913
Nov 21 02:57:52 twx-0 sshd[3912]: debug1: session_exit_message: session 0 channel 0 pid 3913
Nov 21 02:57:52 twx-0 sshd[3912]: debug1: session_exit_message: release channel 0
Nov 21 02:57:52 twx-0 sshd[3912]: Connection closed by 10.133.1.12 port 55398
```
if I use the above code, it failed with the following log:
```
Nov 21 02:58:08 twx-0 sshd[3983]: Starting session: command for dxu from 10.133.1.12 port 55500 id 0
Nov 21 02:58:10 twx-0 sshd[3983]: Corrupted MAC on input.
Nov 21 02:58:10 twx-0 sshd[3983]: ssh_dispatch_run_fatal: Connection from user dxu 10.133.1.12 port 55500: message authentication code incorrect
Nov 21 02:58:10 twx-0 sshd[3983]: debug1: do_cleanup
```
10.133.1.12 is the private IP of proxy box, 10.133.1.5 is the private IP of the target box.
The file size is larger, the failure possibility is higher.
it is reproducible both from windows and linux.

**Comment by vkill** [comment]
v0.3.0 resolved
